manifests/0000_07_cluster-network-operator_03_daemonset.yaml

```diff
@@ -37,6 +37,10 @@ spec:
           value: "quay.io/openshift/origin-container-networking-plugins-supported:v4.0.0"
         - name: CNI_PLUGINS_UNSUPPORTED_IMAGE
           value: "quay.io/openshift/origin-container-networking-plugins-unsupported:v4.0.0"
+        - name: SRIOV_CNI_IMAGE
```

Comment:
These images are too old for KubeVirt / CNV. Is it possible to configure the operator to deploy a different version of the images? Or perhaps disable SR-IOV deployment by this operator so that we can handle deployment of the right components in our own network cluster operator?

This wouldn't be an issue if we would treat OpenShift and upstream versions of the same plugin as different plugins with their own names. Then you would have one CRD for OpenShift with e.g. `type: openshift-sriov` and another with more modern version of the upstream plugin with `type: sriov`. But then we would hit the problem with the old version of the plugin that consumes all VFs found on the node, regardless of wishes of other components (modern plugin can use just part of available VFs). Regardless, it's just not elegant to manually split VFs between two versions of the same plugin, prone to errors and under-utilization.

So the only solution I see is to be able to tell OpenShift operator to not deploy any SR-IOV bits (no CRD, no CNI, no DP), and handle deployment ourselves. Is it something we can consider? If not, it's probably a blocker for CNV SR-IOV integration into OpenShift. Thoughts?

Reply:
I think the cluster operator doesn't control which versions of SR-IOV components are deployed, it's managed by upper layer operator to replace the image version of CNI, DP for upgrading. However, it is possible to disable SR-IOV deployment in network operator, in fact, it's not enabled by default, it will only be deployed when additional network type ``OpenShiftSRIOV`` is specified and corresponding ``SRIOVConfig`` is provided. Re CRD creation, current implementation uses a separate template for SR-IOV CR rendering, it's also triggered when ``OpenShiftSRIOV`` network type is used, but these don't block user creating their own CR with different templates.
With above, I think it's possible for CNV deployment to use a different set of SR-IOV manifests or templates, just don't try to have two sets of SR-IOV components deployed in the same cluster.

This may improve later with OpenShift 4.2, see SR-IOV roadmap [here] (https://docs.google.com/document/d/1lKJjv7ufuMHL_31jPiVEVu6LnlRz8y6Aw-0gV8yY8hc/edit#) when SR-IOV support updates to use latest master.